monai/transforms/croppad/functional.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A collection of "vanilla" transforms for spatial operations
+https://github.com/Project-MONAI/MONAI/wiki/MONAI_Design
+"""
+
+from __future__ import annotations
+
+import numpy as np
+import torch
+
+import monai
+from monai.data.meta_obj import get_track_meta
+from monai.data.meta_tensor import MetaTensor
+from monai.transforms.inverse import TraceableTransform
+from monai.transforms.utils import create_translate
+from monai.utils import TraceKeys, convert_to_dst_type, convert_to_tensor, ensure_tuple
+
+__all__ = ["pad_func", "crop_func"]
+
+
+def pad_func(img, to_pad_, mode, kwargs, transform_info):
+    extra_info = {"padded": to_pad_}
+    img_size = img.peek_pending_shape() if isinstance(img, MetaTensor) else img.shape[1:]
+    spatial_rank = img.peek_pending_rank() if isinstance(img, MetaTensor) else torch.tensor(3.0, dtype=torch.double)
+    do_pad = np.asarray(to_pad_).any()
+    if do_pad:
+        to_pad_ = list(to_pad_)
+        if len(to_pad_) < len(img.shape):
+            to_pad_ = list(to_pad_) + [(0, 0)] * (len(img.shape) - len(to_pad_))
+        to_shift = [-s[0] for s in to_pad_[1:]]  # skipping the channel pad
+        xform = convert_to_dst_type(create_translate(spatial_rank, to_shift), spatial_rank)[0]
```

Comment:
As mentioned before, the `spatial_rank` here needs to be int.
https://github.com/Project-MONAI/MONAI/blob/f4902b220538f02f3621b37c3e1f2067c830e0fe/monai/transforms/utils.py#L860-L861